### PR TITLE
UI :- Added loading state on Save button in Modal

### LINKED ISSUE #7198: UI: Improve UX on API calls from a modal dialog
When API calls are made from a modal dialog for example in an edit description of an entity flow,  when the user clicks on the "Save" button, the dialog box closes immediately regardless of the response status.

This will be bad UX if the API call fails for some reason since the user has to type in the lengthy description again and try to save again. 

Expected Behavior:

- Disable the "Save" button, and show a loading indicator when the API call is being made.
- If the response is success, change the loading indicator to a green check mark and then close the modal dialog 
- If the response if failure, show the error message in the modal dialog itself and enable the "Save" button for the user to retry

https://user-images.githubusercontent.com/1051198/188285481-ed9cb088-9cc8-44d2-8003-6082c9d63c02.mov

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes :

- Issue :- Closes #7198
- Modal will be open if error occur.
- Added loading state onSave click.

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/66266464/196196185-1e930e7f-07e6-4560-a831-c1693323599c.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/AddDataQualityTest/EditTestCaseModal.tsx [hunk 1]
```diff
@@ -74,9 +78,9 @@ const EditTestCaseModal: React.FC<EditTestCaseModalProps> = ({
           <Form.Item
             data-testid="sql-editor-container"
             key={name}
-            label="SQL Query"
+            label={t('label.sql-query')}
             name={name}
-            tooltip="Queries returning 1 or more rows will result in the test failing.">
+            tooltip={t('label.queries-failing')}>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/AddDataQualityTest/EditTestCaseModal.tsx:83]:
```suggestion
            tooltip={t('label.sql-query-tooltip')}>
```

File: openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx [hunk 2]
```diff
@@ -213,8 +219,6 @@ const AuthMechanismForm: FC<Props> = ({
       }
     } catch (error) {
       showErrorToast(error as AxiosError);
-    } finally {
-      setIsConfirmationModalOpen(false);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx:217]: Can you explain why we removed this and why we are not marking it as false anywhere else

File: openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx [hunk 3]
```diff
@@ -388,79 +392,79 @@ const AuthMechanismForm: FC<Props> = ({
...
             <Form.Item
-              label="Email"
+              label={t('label.email')}
               name="oktaEmail"
               rules={[
                 {
                   required: true,
                   type: 'email',
-                  message: 'Service account Email is required',
+                  message: t('label.service-email-required'),
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx:452]: this should be part of message & not label

Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx:452]: Please update it at every place

File: openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx [hunk 4]
```diff
@@ -570,14 +574,14 @@ const AuthMechanismForm: FC<Props> = ({
...
-            label="Token Expiration"
+            label={t('label.token-expiration')}
             name="token-expiration"
             rules={[
               {
                 required: true,
                 validator: () => {
                   if (!tokenExpiry) {
-                    return Promise.reject('Token Expiration is required');
+                    return Promise.reject(t('label.token-expiration-required'));
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx:584]: message

File: openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx [hunk 5]
```diff
@@ -640,13 +644,15 @@ const AuthMechanismForm: FC<Props> = ({
...
+          confirmLoading={isLoading}
+          okText={t('label.confirm')}
+          title={t('label.are-you-sure')}
           visible={isConfirmationModalOpen}
           onCancel={() => setIsConfirmationModalOpen(false)}
           onOk={handleAccountEmailChange}>
           <Typography.Text>
-            {BOT_ACCOUNT_EMAIL_CHANGE_CONFIRMATION} for {botData.name} bot.
+            {BOT_ACCOUNT_EMAIL_CHANGE_CONFIRMATION} {t('label.for')}{' '}
+            {botData.name} {t('label.bot')}.
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx:655]: It should be a single message on message.bot-email success with providing botName
Reference: https://www.i18next.com/translation-function/interpolation

File: openmetadata-ui/src/main/resources/ui/src/components/Modals/KillIngestionPipelineModal/KillIngestionPipelineModal.tsx [hunk 6]
```diff
@@ -41,15 +43,12 @@ const KillIngestionModal: FC<KillIngestionModalProps> = ({
       const status = response.status;
       if (status === 200) {
         onClose();
-        showSuccessToast(
-          `Successfully killed running workflows for ${pipelinName}.`
-        );
+        showSuccessToast(` ${t('label.kill-successfully')}  ${pipelinName}.`);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Modals/KillIngestionPipelineModal/KillIngestionPipelineModal.tsx:46]: message

File: openmetadata-ui/src/main/resources/ui/src/components/Modals/KillIngestionPipelineModal/KillIngestionPipelineModal.tsx [hunk 7]
```diff
@@ -62,13 +61,12 @@ const KillIngestionModal: FC<KillIngestionModalProps> = ({
...
       okText="Confirm"
-      title={`Kill ${pipelinName} ?`}
+      title={`${t('label.kill')} ${pipelinName} ?`}
       visible={isModalOpen}
       onCancel={onClose}
       onOk={handleConfirm}>
       <Typography.Text data-testid="kill-modal-body">
-        Once you kill this Ingestion, all running and queued workflows will be
-        stopped and marked as Failed.
+        {t('label.kill-ingestion')}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Modals/KillIngestionPipelineModal/KillIngestionPipelineModal.tsx:69]: message

File: openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx [hunk 8]
```diff
@@ -184,17 +190,18 @@ const ProfilerSettingsModal: React.FC<ProfilerSettingsModalProps> = ({
...
+      okText={t('label.save')}
+      title={t('label.settings')}
       visible={visible}
       width={630}
       onCancel={handleCancel}
       onOk={handleSave}>
       <Row gutter={[16, 16]}>
         <Col data-testid="profile-sample-container" span={24}>
-          <p>Profile Sample %</p>
+          <p>{t('label.profile-sample')}</p>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/TableProfiler/Component/ProfilerSettingsModal.tsx:204]:
```suggestion
          <p>{t('label.profile-sample-percentage')}</p>
```

**Review by chirag-madlani — COMMENTED** [review]
@Ashish8689 Let's handle localisation as I suggested in few files and then we can proceed further

File: openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx (hunk 2 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx:217]: @chirag-madlani  we are making it in handleBotUpdate in try block on line number 179.

File: openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx [hunk 9]
```diff
@@ -464,7 +464,7 @@ const AuthMechanismForm: FC<Props> = ({
               <Input
                 data-testid="scopes"
                 name="scopes"
-                placeholder={t('label.scopes-comma-separated')}
+                placeholder={t('message.scopes-comma-separated')}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/BotDetails/AuthMechanismForm.tsx:467]: Placeholder will be still label
Error message will be message

File: openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json [hunk 10]
```diff
@@ -122,51 +122,53 @@
...
+    "clientId-required": "ClientId is required",
+    "authority-required": "Authority is required",
+    "scopes-required": "Scopes is required",
+    "scopes-comma-separated": "Scopes value comma separated",
+    "secretKey-required": "SecretKey is required",
+    "tokenEndpoint-required": "TokenEndpoint is required",
+    "privateKey-required": "PrivateKey is required",
+    "auth-mechanism-required": "Auth Mechanism is required",
+    "kill-successfully": "Successfully killed running workflows for",
+    "kill-ingestion": "Once you kill this Ingestion, all running and queued workflows will be stopped and marked as Failed.",
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json:169]: "kill-ingestion-warning"

**Comment by Ashish8689** [comment]
@chirag-madlani  localisation label updated!

**Review by chirag-madlani — DISMISSED** [review]

**Pull request merged**